src/provider/element-templates/util/templateUtil.js

```diff
@@ -89,3 +94,30 @@ function leftPad(string, length, character) {
 
   return string;
 }
+
+function createBlankBusinessObject(element, injector) {
+  const bpmnFactory = injector.get('bpmnFactory'),
+        bpmnJs = injector.get('bpmnjs');
+
+  const bo = getBusinessObject(element),
+        newBo = bpmnFactory.create(bo.$type),
+        label = getLabel(element);
+
+  if (!label) {
+    return newBo;
+  }
+
+  if (is(element, 'bpmn:TextAnnotation')) {
```

Comment:
You could use [`LabelUtil#setLabel`](https://github.com/bpmn-io/bpmn-js/blob/develop/lib/features/label-editing/LabelUtil.js#L52) here.

Reply:
And wrap it in a `try/catch` for good measure to handle categoryRef not existing, of course.

Reply:
> And wrap it in a try/catch for good measure to handle categoryRef not existing, of course

Ah, slow Github updates :smile:

Reply:
Done in [4b961e0](https://github.com/bpmn-io/bpmn-js-properties-panel/pull/662/commits/4b961e092266ca77bb9784b3faf88818c67b44a0)